import_json.go

```diff
@@ -110,26 +115,46 @@ func main() {
 		if _, ok := cachedIds[id]; ok {
 			continue
 		}
+		newTweetsID = append(newTweetsID, id)
+	}
 
-		time.Sleep(1 * time.Second)
-		tweet, resp, err := client.Statuses.Show(id, nil)
-		if err != nil {
-			log.Printf("failed to get tweet %s: %v\n", text, err)
-			continue
-		}
-		if resp.StatusCode != 200 {
-			log.Printf("failed to get tweet %s: status=%d, %v\n", text, resp.StatusCode, err)
-			continue
-		}
+	tweets, err := fetchTweets(client, newTweetsID)
+	if err != nil {
+		log.Fatalf("failed to get tweet: %v\n", err)
+	}
 
+	for _, tweet := range tweets {
 		tweetJson, _ := json.Marshal(tweet)
 		fmt.Println(string(tweetJson))
 		fmt.Fprintln(w, string(tweetJson))
+		if err := stdin.Err(); err != nil {
+			log.Fatalln(err)
+		}
 	}
-	if err := stdin.Err(); err != nil {
-		log.Fatalln(err)
-	}
+
 	if err := w.Sync(); err != nil {
 		log.Fatalf("failed to flush tweets: %v\n", err)
 	}
 }
+
+func fetchTweets(client *twitter2.Client, ids []string) (tweets []*twitter2.Tweet, err error) {
+	// the `tweets` API accepts less than 100 tweets.
+	const N = 100
+	c := 0
+	for start := 0; start < len(ids); start += N {
+		end := start + N
+		if end > len(ids) {
+			end = len(ids)
+		}
+		t, err := client.Tweets(ids[start:end])
+		if err != nil {
+			return nil, err
+		}
+		c += len(t)
+		log.Printf("fetched %d tweets...", c)
+		tweets = append(tweets, t...)
+		// avoid `too many requests`
+		time.Sleep(time.Minute)
+	}
+	return
```

Comment:
なるほど
> Don't name result parameters just to avoid declaring a var inside the function; that trades off a minor implementation brevity at the cost of unnecessary API verbosity.